Feature/issue 83/create database to store difficulty level for allostatic regulation

sdubey2024 opened the pull request:
- added python script to create user settings database for storing allostatic filter strength 
- updated files in databases folder to include new database 
- removed metadata from categorised database

sdubey2024 committed:
Merge pull request #85 from mouper/main

merge changes in main to new branch

sdubey2024 committed:
feat: create user settings db

Contains the filter_strength for allostatic regulation

sdubey2024 committed:
feat: add db path to config.py

Add the user_settings_db path to the config file and correct all paths

sdubey2024 committed:
feat: update database.py

Remove metadata from categorised database and add functions for user settings database

sdubey2024 committed:
feat: update test.py

Update paths for all databases, remove metadata from categorised database, add testing code for user settings database.

sdubey2024 committed:
feat: update watched_db.py

Set video_id as primary key to identify unique rows

mouper reviewed (changes requested):
Nice work Sia, just a few things I want to confirm/ think could be changed:

1. The filter strength options in 'filter_strength TEXT CHECK(filter_strength IN ('LOW', 'MEDIUM', 'HARD'))' should be Weak, Moderate, Strong and Extra Strong so it aligns with the dashboard design the client approved.
2. The database mentioned in this issue is a different database to the one created, as we discussed in Discord, there were 2 new databases to be added: the user settings db (that you have added in this PR) and one for the allostatic graph data maybe named something like graph_data.db 
- Is this issue meant to include adding both of these or is it meant to just be the user_settings? If this is the case, I think it would be easier to add the other database in within this pull request as well, rather than doing another issue/ PR for it.

sdubey2024 committed:
feat: change filter strength values

Changed filter strength values to Weak, Moderate, Strong and Extra Strong in user settings database

sdubey2024 committed:
feat: update user settings db

Changed filter strength values to Weak, Moderate, Strong and Extra Strong in user settings database

sdubey2024 commented:
I have changed the filter strength options. This issue and pull request is only for the user settings database. I have created a separate issue for the graph data db.

sdubey2024 commented:
It will be easier for me to merge this pull request and then start working on the other database.

mouper reviewed (approved):
Sure, looks good to me then!
